custom_components/places/sensor.py

```diff
@@ -895,6 +896,10 @@ def get_zone_details(self):
                 self.set_attr(
                     ATTR_DEVICETRACKER_ZONE_NAME, self.get_attr(ATTR_DEVICETRACKER_ZONE)
                 )
+                if self._hass.states.get("zone." + str(self.get_attr(ATTR_DEVICETRACKER_ZONE))) is not None and self._hass.states.get("zone." + str(self.get_attr(ATTR_DEVICETRACKER_ZONE))).attributes.get(CONF_FRIENDLY_NAME) is not None:
+                    self.set_attr(
+                        ATTR_DEVICETRACKER_ZONE_NAME, self._hass.states.get("zone." + str(self.get_attr(ATTR_DEVICETRACKER_ZONE))).attributes.get(CONF_FRIENDLY_NAME))
+
```

Comment:
Will add a commit that does this, but I think this needs to be moved in the "if devicetracker_zone_name_state is not None:" section.